shift-hold is error-prone; try shift-back

NomadicVolcano opened the pull request:
Using shift-HOLD has been extremely error-prone in my tests.  Try shift-BACK instead.  These changes will automatically switch between STOP and reverse thrust when you push and release the SHIFT key.

NomadicVolcano committed:
"stop command" is shift-down instead of shift-hold

petervdmeer commented on source/Engine.cpp:
```diff
@@ -1608,22 +1608,22 @@ void Engine::HandleKeyboardInputs()
 	// Transfer all newly pressed unhandled keys to active commands, except for HOLD, since
 	// HOLD has a different meaning depending on if shift was pressed (and is handled earlier).
-	activeCommands |= keyDown.AndNot(Command::HOLD);
+	activeCommands |= keyDown;
```
If I would press SHIFT, and then also press BACK, would this then cause the BACK command to be transferred for 1 cycle as well (besides the STOP command)?

NomadicVolcano replied:
That was a bug.  It should have `.AndNot(Command::BACK)`

petervdmeer commented on source/Engine.cpp:
```diff
@@ -1608,22 +1608,22 @@ void Engine::HandleKeyboardInputs()
 		Command::SHIFT));
 	
 	// Transfer all newly pressed unhandled keys to active commands, except for HOLD, since
 	// HOLD has a different meaning depending on if shift was pressed (and is handled earlier).
```
This remark about HOLD no longer seems relevant (since the meaning of Hold just again became what it was).

NomadicVolcano replied:
Comment is updated to reflect that BACK is handled differently instead of HOLD

Commit:
Bug fix for this problem:

1. user presses and holds SHIFT
2. user presses and holds BACK
3. ship engages reverse thrusters (BACK) for one step
4. ship disengages reverse thrusters and then activates STOP
